packages/hooks/src/useGrowthbookFeatureFlag.ts

```diff
@@ -1,28 +1,45 @@
 import { useState, useEffect } from 'react';
 import { Analytics } from '@deriv-com/analytics';
+import { useRemoteConfig } from '@deriv/api';
 
-interface UseGrowthbookFeatureFlagArgs<T> {
+interface UseGrowthbookFeatureFlagArgs {
     featureFlag: string;
-    defaultValue: T;
 }
 
-const useGrowthbookFeatureFlag = <T extends string | boolean>({
-    featureFlag,
-    defaultValue,
-}: UseGrowthbookFeatureFlagArgs<T>) => {
-    const [featureFlagValue, setFeatureFlagValue] = useState<T>(defaultValue);
+const useGrowthbookFeatureFlag = ({ featureFlag }: UseGrowthbookFeatureFlagArgs) => {
+    const [featureFlagValue, setFeatureFlagValue] = useState(Analytics?.getFeatureValue(featureFlag));
+    const { data } = useRemoteConfig();
 
     useEffect(() => {
-        const value = (Analytics?.getFeatureValue(featureFlag, defaultValue) || defaultValue) as T;
-        setFeatureFlagValue(value);
-
-        // Set the renderer for GrowthBook to update the value when the feature flag changes
-        Analytics.getInstances()?.ab?.GrowthBook?.setRenderer(() => {
-            const value = (Analytics?.getFeatureValue(featureFlag, defaultValue) || defaultValue) as T;
-
-            setFeatureFlagValue(value);
-        });
-    }, [featureFlag, defaultValue]);
+        let analytics_interval: NodeJS.Timeout;
+        if (data.marketing_growthbook) {
+            let timeout = 0;
+            analytics_interval = setInterval(() => {
+                // Check if the analytics instance is available for 10 seconds before setting the feature flag value
+                if (timeout > 20) {
+                    // If the analytics instance is not available after 10 seconds, clear the interval
+                    clearInterval(analytics_interval);
+                    return;
+                }
```

Comment:
here also we need undefined check. isnt? Analytics?.getInstances()?.ab

Reply:
done